namer/dnssrv/src/main/scala/io/buoyant/namer/dnssrv/DnsSrvNamer.scala

```diff
@@ -103,13 +111,13 @@ class DnsSrvNamer(
             log.error(msg)
             update.update(Activity.Failed(new DNSLookupException(msg)))
         }
+        counter = counter + 1
       }
-
-      doUnit()
-
-      timer.schedule(refreshInterval) {
+      val task: TimerTask = timer.schedule(refreshInterval) {
         doUnit()
       }
+      doUnit()
+      task
```

Comment:
I would keep the existing order here to ensure that we only schedule subsequent runs once the first one has completed.